apps/condo/domains/notification/schema/MarketingMessage.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Generated by `createschema notification.MarketingMessage 'type:Select:sms,email,push; message:Text; deepLink?:Text; idList:Json; status:Select:created,done; processingMeta?:Json;'`
+ */
+
+const { isArray, isEmpty } = require('lodash')
+
+const { Text, Select } = require('@keystonejs/fields')
+const { Json } = require('@condo/keystone/fields')
+const { GQLListSchema } = require('@condo/keystone/schema')
+const { historical, versioned, uuided, tracked, softDeleted } = require('@condo/keystone/plugins')
+
+const { dvAndSender } = require('@condo/domains/common/schema/plugins/dvAndSender')
+const { hasValidJsonStructure } = require('@condo/domains/common/utils/validation.utils')
+const access = require('@condo/domains/notification/access/MarketingMessage')
+
+const { JSON_EXPECT_ARRAY_ERROR, OPERATION_FORBIDDEN } = require('@condo/domains/common/constants/errors')
+
+const {
+    MARKETING_MESSAGE_TRANSPORT_TYPES,
+    MARKETING_MESSAGE_STATUSES,
+    MARKETING_MESSAGE_CREATED_STATUS,
+} = require('../constants/constants')
+
+const { sendMarketingMessagesTask } = require('../tasks/sendMarketingMessagesTask')
+
+const operationForbiddenValidator = ({ fieldPath, addFieldValidationError, operation }) => {
+    if (operation === 'update') addFieldValidationError(`${OPERATION_FORBIDDEN}] Updating ${fieldPath} is forbidden.`)
+}
+
+const MarketingMessage = new GQLListSchema('MarketingMessage', {
+    schemaDoc: 'Marketing message batch. Adding record here will cause sending message to all users listed in idList via transport, selected in transportType.',
+    fields: {
+
+        transportType: {
+            schemaDoc: 'Transport type to send notifications by.',
+            type: Select,
+            options: MARKETING_MESSAGE_TRANSPORT_TYPES,
+            isRequired: true,
+            hooks: {
+                validateInput: operationForbiddenValidator,
+            },
+        },
+
+        title: {
+            schemaDoc: 'Title for message to send.',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                validateInput: operationForbiddenValidator,
+            },
+        },
+
+        message: {
+            schemaDoc: 'Message to send.',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                validateInput: operationForbiddenValidator,
+            },
+        },
+
+        deepLink: {
+            schemaDoc: 'Used for push notifications only. Useful for moving user to proper position within mobile apps.',
+            type: Text,
+            hooks: {
+                validateInput: operationForbiddenValidator,
+            },
+        },
+
+        idList: {
```

Comment:
probably it should be: ID list (for PUSH/EMAIL/SMS) / email list (for Email) / phone list (for SMS)